src/lang/base/Utils.ml

```diff
@@ -51,7 +55,7 @@ module AssocDictionary : Dictionary = struct
   let rec remove k d =
     match d with
     | []              -> []
-    | (kd, vd) :: rest -> if k = kd then rest else (kd, vd) :: (remove k rest)
+    | (kd, vd) :: rest -> if k = kd then (remove k rest) else (kd, vd) :: (remove k rest)
```

Comment:
This change will break the cashflow analyser in the case where two variables have the same name.